Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/AWSCredentialIdentityResolvers/STSAssumeRoleAWSCredentialIdentityResolver.swift

```diff
@@ -48,3 +50,9 @@ public struct STSAssumeRoleAWSCredentialIdentityResolver: AWSCredentialIdentityR
 }
 
 // swiftlint:enable type_name
+
+func validateString(name: String, regex: String) throws {
+    guard let _ = name.range(of: regex, options: .regularExpression) else {
```

Comment:
`name(range:options:)` is an old method inherited from ObjC.

See here for Swift-native, modern regex support:
https://developer.apple.com/documentation/swift/regex

Reply:
After review, Swift regex features require platform support beyond our minimums.  It's okay to keep Foundation-based regex support in place for now.